Modules/_sqlite/statement.c

```diff
@@ -240,11 +239,9 @@ void pysqlite_statement_bind_parameters(pysqlite_Statement* self, PyObject* para
         }
         for (i = 0; i < num_params; i++) {
             if (PyTuple_CheckExact(parameters)) {
-                current_param = PyTuple_GET_ITEM(parameters, i);
-                Py_INCREF(current_param);
+                current_param = Py_NewRef(PyTuple_GET_ITEM(parameters, i));
             } else if (PyList_CheckExact(parameters)) {
-                current_param = PyList_GetItem(parameters, i);
-                Py_XINCREF(current_param);
+                current_param = Py_XNewRef(PyList_GetItem(parameters, i));
```

Comment:
It looks strange that the list case accepts NULL but not the the tuple case. I expect that it cannot be NULL.

Reply:
Apparently it can:
```
./python.exe -m test -F -r -j1 -R 3:5 test_sqlite
Using random seed 2375104
0:00:00 load avg: 3.56 Run tests in parallel using 1 child processes
0:00:00 load avg: 3.51 [  1/1] test_sqlite crashed (Exit code -11)
beginning 8 repetitions
12345678
Fatal Python error: Segmentation fault

Current thread 0x0000000107fbcdc0 (most recent call first):
  File "/Users/erlendaasland/src/cpython.git/Lib/sqlite3/test/regression.py", line 146 in CheckBindMutatingList
```